### ISSUE TITLE
get_distribution() does not accept np.int64 int as argument

### ISSUE THREAD

**Issue by wdeback** [neuralyzer/pyabc#2]
I'm trying to get the weighted average of the parameters for each population in an ABC run, like this: 

```
for pop in history.get_all_populations()['t'].unique():
    if pop < 0: 
        continue
    df_params, weights = history.get_distribution(m=0, t=pop)   # <-- pop is of type 'numpy.int64'
    mean_params = np.average(df_params.values, weights=weights, axis=0)
    print('population: {}, params: {}'.format(pop, mean_params))
```
However, this gives an error: `ZeroDivisionError: Weights sum to zero, can't be normalized`.

This can be solved by casting the argument to get_distribution to int like this (note the `int(pop)` argument: 

`    df_params, weights = history.get_distribution(m=0, t=int(pop))   # <-- pop is of type 'int'
`

Thus, it appears that get_distribution() does not accept numpy.int64 as a valid argument although it is a valid integer. Perhaps this casting can be done internally? Alternatively, a TypeError exception should be raised.

**Comment by neuralyzer** [comment]
Thanks for reporting.

I added a test for this kind of behaviour (test.test_strorage.test_single_particle_save_load_np_int64) and changed the code accordingly. 
Can you confirm that this fixed the bug?

**Comment by wdeback** [comment]
Fix in 7419e6e817f9ef21d4edaa141bd6dbe6e55ab7e5 confirmed.

**Comment by neuralyzer** [comment]
Thanks. I added the same fix at a couple more places. You might want to pull again.
